iModelCore/ECDb/ECDb/ECSchemas/ECDbMeta.04.00.02.ecschema.xml

```diff
@@ -0,0 +1,669 @@
+    </ECEnumeration>
+    <ECEnumeration typeName="ECClassType" backingTypeName="int" isStrict="true">
+        <ECEnumerator name="Entity" value="0" displayLabel="Entity" />
+        <ECEnumerator name="Relationship" value="1" displayLabel="Relationship" />
+        <ECEnumerator name="Struct" value="2" displayLabel="Struct" />
+        <ECEnumerator name="CustomAttribute" value="3" displayLabel="CustomAttribute" />
+    </ECEnumeration>
+    <ECEnumeration typeName="ECPropertyKind" backingTypeName="int" isStrict="true">
+        <ECEnumerator name="Primitive" value="0" displayLabel="Primitive" />
+        <ECEnumerator name="Struct" value="1" displayLabel="Struct" />
+        <ECEnumerator name="PrimitiveArray" value="2" displayLabel="PrimitiveArray" />
+        <ECEnumerator name="StructArray" value="3" displayLabel="StructArray" />
+        <ECEnumerator name="Navigation" value="4" displayLabel="Navigation" />
+    </ECEnumeration>
+    <ECEnumeration typeName="ECRelationshipDirection" backingTypeName="int" isStrict="true">
+        <ECEnumerator name="Forward" value="1" displayLabel="Forward" />
+        <ECEnumerator name="Backward" value="2" displayLabel="Backward" />
+    </ECEnumeration>
+    <ECEnumeration typeName="ECRelationshipStrength" backingTypeName="int" isStrict="true">
+        <ECEnumerator name="Referencing" value="0" displayLabel="Referencing" />
+        <ECEnumerator name="Holding" value="1" displayLabel="Holding" />
+        <ECEnumerator name="Embedding" value="2" displayLabel="Embedding" />
+    </ECEnumeration>
+    <ECEnumeration typeName="PrimitiveType" backingTypeName="int" isStrict="true">
+        <ECEnumerator name="Binary" value="257" displayLabel="Binary" />
+        <ECEnumerator name="Boolean" value="513" displayLabel="Boolean" />
+        <ECEnumerator name="DateTime" value="769" displayLabel="DateTime" />
+        <ECEnumerator name="Double" value="1025" displayLabel="Double" />
+        <ECEnumerator name="Integer" value="1281" displayLabel="Integer" />
+        <ECEnumerator name="Long" value="1537" displayLabel="Long" />
+        <ECEnumerator name="Point2d" value="1793" displayLabel="Point2d" />
+        <ECEnumerator name="Point3d" value="2049" displayLabel="Point3d" />
+        <ECEnumerator name="String" value="2305" displayLabel="String" />
+        <ECEnumerator name="IGeometry" value="2561" displayLabel="IGeometry" />
+    </ECEnumeration>
+
+    <!-- Flat list of Custom Attributes without relationships to their container -->
+    <ECEntityClass typeName="CustomAttribute" description="" displayLabel="" modifier="Sealed">
+        <ECCustomAttributes>
+            <ClassMap xmlns="ECDbMap.02.00.00">
+                <MapStrategy>ExistingTable</MapStrategy>
+                <TableName>ec_CustomAttribute</TableName>
+            </ClassMap>
+        </ECCustomAttributes>
+        <ECProperty propertyName="ContainerId"  typeName="int"/>

```

Comment:
If this is the Id of the Class, property or whatever shouldn't this be a long instead of an int?

We are kind of cheating here by making this a number instead of an explicit id.  So we should probably use an extended type here to give at least some hint.

Reply:
yes, we should use `extendedType="Id"` otherwise serializer fails to convert it into hex.

Reply:
done